src/components/TicTacToe.tsx

```diff
@@ -25,10 +31,16 @@ class TicTacToe extends React.Component<object, GameState> {
 
   public submitBoardSize = (event: React.FormEvent<HTMLFormElement>): void => {
     event.preventDefault();
-    this.game = new Game(this.state.boardSize);
-    this.setState({
-      board: this.game.board,
-    });
+    if (this.validateBoardSizeInput(this.state.boardSize)) {
```

Comment:
If this method is a function of internal state, there's no need to pass that state around internally; everything already has access to it.